framework/include/base/MooseApp.h

```diff
@@ -616,6 +571,9 @@ class MooseApp :
 
 private:
 
+  /// pointer to the parent/owning MultiApp object if any.
+  MultiApp * _parent_multi = nullptr;
```

Comment:
The MultiApp design has been successful precisely because of the compartmentalization of the sub-apps.  They are not supposed to even be aware that they are running as a sub-app.  It provides clear boundaries and makes it easier to piece together complex simulations.  It's designed this way on purpose.

If we were to start growing "links" upwards... everything would get entangled in such a mess that we would never be able to do things like prototype individual solves by themselves and then piece them together into a bigger solve... which is what we can do with MultiApps right now.